frontend/packages/noobaa-storage-plugin/src/components/status-card/statuses.tsx

```diff
@@ -0,0 +1,92 @@
+import * as _ from 'lodash';
+import { HealthState } from '@console/shared/src/components/dashboard/health-card/states';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+import { FirehoseResult } from '@console/internal/components/utils';
+import { getGaugeValue } from '../../utils';
+
+const NooBaaStatus = [
+  {
+    state: HealthState.ERROR,
+    message: 'MCG is not running',
+  },
+  {
+    state: HealthState.ERROR,
+    message: 'All resources are unhealthy',
+  },
+  {
+    state: HealthState.WARNING,
+    message: 'Object Bucket has an issue',
+  },
+  {
+    state: HealthState.ERROR,
+    message: 'Many buckets have issues',
+  },
+  {
+    state: HealthState.WARNING,
+    message: 'Some buckets have issues',
+  },
+];
+
+export const getNooBaaState: GetObjectServiceStatus = (
+  prometheusResponses = [],
+  hasLoadError,
+  isLoading,
+  k8sResponse,
+) => {
+  const [buckets, unhealthyBuckets, pools, unhealthyPools] = prometheusResponses.map((r) =>
+    getGaugeValue(r),
+  );
+  const noobaaPhase = _.get(k8sResponse, 'data[0].status.phase');
+  const unhealthyBucketsRatio = unhealthyBuckets / buckets;
+
+  if (hasLoadError) {
+    return { state: HealthState.UNKNOWN };
+  }
+  if (isLoading) {
+    return { state: HealthState.LOADING };
+  }
+  if (noobaaPhase !== 'Ready') {
+    return NooBaaStatus[0];
+  }
+  if (pools && unhealthyPools && Number(pools) === Number(unhealthyPools)) {
```

Comment:
```suggestion
  if (Number(pools) === Number(unhealthyPools)) {
```
Please update `getGaugeValue` to not return `null`. `undefined` is much better to use when we are dealing with numeric values. However that is being changed by 2958 too.